src/modalities/nn/model_initialization/initialization_routines.py

```diff
@@ -0,0 +1,122 @@
+import math
+import re
+from typing import Annotated, List, Optional
+
+import torch.nn as nn
+from pydantic import BaseModel, Field, model_validator
+
+from modalities.nn.model_initialization.initialization_if import ModelInitializationIF
+from modalities.nn.model_initialization.parameter_name_filters import RegexFilter
+
+
+class PlainInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    parameter_name_regexes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+    hidden_dim: Optional[int] = None
+
+    @model_validator(mode="after")
+    def check_std_and_hidden_dim(self):
+        if self.std == "auto" and self.hidden_dim is None:
+            raise ValueError("hidden_dim must be specified when std is 'auto'")
+        return self
+
+
+class ScaledInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)]
+    num_layers: Annotated[int, Field(strict=True, gt=0)]
+    parameter_name_regexes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class ScaledEmbedInitializationConfig(BaseModel):
+    mean: float
+    parameter_name_regexes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class NamedParameterwiseNormalInitialization(ModelInitializationIF):
+    def __init__(self, mean: float, std: float, parameter_name_regexes: RegexFilter):
+        self.mean = mean
+        self.std = std
+        self.parameter_name_regexes = parameter_name_regexes
+
+    def initialize_in_place(self, model: nn.Module):
+        for parameter_name, p in model.named_parameters():
+            weight_regexes = self.parameter_name_regexes.weights
+            for weight_regex in weight_regexes:
+                if re.fullmatch(weight_regex, parameter_name):
+                    nn.init.normal_(p, mean=self.mean, std=self.std)
+            bias_regexes = self.parameter_name_regexes.biases
+            for bias_regex in bias_regexes:
+                if re.fullmatch(bias_regex, parameter_name):
+                    nn.init.zeros_(p)
+
+
+class InitializationRoutines:
+    @staticmethod
+    def get_plain_initialization(
+        mean: float, std: float | str, parameter_name_regexes: List[str], hidden_dim: Optional[int] = None
+    ) -> NamedParameterwiseNormalInitialization:
```

Comment:
```suggestion
    ) -> ModelInitializationIF
```
To specify the abstract base class instead of the more specific inherited class here is in accordance with what is done for all related methods (`get_scaled_initialization`, `get_scaled_embed_initialization`, `get_composed_model_initializer`, `get_model_initializer_wrapper`).

Reply:
Generally true, but we need the concrete class in `ComposedInitializationRoutines.get_composed_model_initializer(...)`, as we need to access `plain_init.std`. Otherwise, there is no typing support. 

https://github.com/Modalities/modalities/blob/eea99bc1ba86923c8e9fbfa410c8bc502e8598f4/src/modalities/nn/model_initialization/composed_initialization.py#L116-L120